cmd/test-otel-init-go/main_test.go

```diff
@@ -0,0 +1,291 @@
+		t.Errorf("env data did not match fixture in %q (-want +got):\n%s", scenario.Filename, diff)
+	}
+
+	// check the otel-init-go config
+	if diff := cmp.Diff(scenario.StubData.Config, stubData.Config); diff != "" {
+		t.Errorf("config data did not match fixture in %q (-want +got):\n%s", scenario.Filename, diff)
+	}
+
+	// check the otel span values
+	// find usages of *, do the check on the stub data manually, and set up cmpSpan
+	scSpan := map[string]string{}  // to be passed to cmp.Diff
+	cmpSpan := map[string]string{} // to be passed to cmp.Diff
+	for what, re := range map[string]*regexp.Regexp{
+		"trace_id":    regexp.MustCompile("^[0-9a-fA-F]{32}$"),
+		"span_id":     regexp.MustCompile("^[0-9a-fA-F]{16}$"),
+		"is_sampled":  regexp.MustCompile("^true|false$"),
+		"trace_flags": regexp.MustCompile("^[0-9]{2}$"),
+	} {
+		if cv, ok := scenario.StubData.Otel[what]; ok {
+			scSpan[what] = cv // make a straight copy to make cmp.Diff happy
+			if sv, ok := stubData.Otel[what]; ok {
+				cmpSpan[what] = sv // default to the existing value
+				if cv == "*" {
+					if re.MatchString(sv) {
+						cmpSpan[what] = "*" // success!, make the Cmp test succeed
+					} else {
+						t.Errorf("stub span value %q for key %s is not valid", sv, what)
+					}
+				}
+			}
+		}
+	}
+
+	// do a diff on a generated map that sets values to * when the * check succeeded
+	if diff := cmp.Diff(scSpan, cmpSpan); diff != "" {
+		t.Errorf("otel data did not match fixture in %q (-want +got):\n%s", scenario.Filename, diff)
+	}
+}
+
+// runPrograms runs the stub program and otel-cli together and captures their
+// output as data to return for further testing.
+// all failures are fatal, no point in testing if this is broken
+func runPrograms(t *testing.T, scenario Scenario) (StubData, CliEvents) {
+	tmpdir, err := os.MkdirTemp(os.TempDir(), "otel-init-go-test")
+	defer os.RemoveAll(tmpdir)
+	if err != nil {
+		t.Fatalf("MkdirTemp failed: %s", err)
+	}
+
+	cliArgs := []string{"server", "json", "--dir", tmpdir}
+
+	if scenario.Timeout > 0 {
+		cliArgs = append(cliArgs, "--timeout", strconv.Itoa(scenario.Timeout))
+	}
+
+	if scenario.SpansExpected > 0 {
+		cliArgs = append(cliArgs, "--max-spans", strconv.Itoa(scenario.SpansExpected))
+	}
+
+	// MAYBE: server json --stdout is maybe better? and could add a graceful exit on closed fds
+	otelcli := exec.Command(otelCliPath, cliArgs...)
+	otelcli.Env = []string{"PATH=/bin"} // apparently this is required for 'getent', no idea why
+
+	if !scenario.SkipOtelCli {
+		go func() {
+			err, output := otelcli.CombinedOutput()
+			if err != nil {
+				log.Println(output)
+				log.Fatalf("Executing command %q failed: %s", otelcli.String(), err)
+			}
+		}()
+	}
+
+	// TODO: replace this with something more reliable
+	// the problem here is we need to wait for otel-cli to start and listen
+	// so the solution is probably to do some kind of healthcheck on otel-cli's port
+	// but this works ok for now
+	time.Sleep(time.Millisecond * 10)
```

Comment:
https://github.com/equinix-labs/otel-cli/pull/51 splits the otel-cli server into a reusable package with a little more control over startup so this can go away soon and it gets rid of needing to download otel-cli in the build pipeline which is kinda nice too.